azure - container host handles attribute errors while unloading

stefangordon opened the pull request:
We examine policies that we unload so that we can do cleanup. 

I was able to find a few ways to sabotage policies such that they'd throw an attribute error in that code - this is a quick fix to go ahead and delete them anyway if you end up in that situation.  

This method could probably be refactored a bit, but trying to avoid testing overhead and get a fix in.  I've been using this fix on another branch for a few days.

#5735

kapilt commented on tools/c7n_azure/c7n_azure/container_host/host.py:
```diff
@@ -258,22 +258,24 @@ def unload_policy_file(self, path, policies):
-            self.scheduler.remove_job(job_id=name)
+            for name in periodic_to_remove:
+                self.scheduler.remove_job(job_id=name)
+        except AttributeError as exc:
```
the error in the referenced issue is a keyerror instead of attribute

stefangordon replied:
We should probably catch anything here I suppose.  If we manage to miss a cleanup it’ll get resolved later harmlessly, best effort and then delete the file.

stefangordon commented:
Ok the two failure modes I found were a policy that was never loaded for any reason (didn't validate), and a YAML file without a `policies` element.  You could get an `AttributeError` or a `KeyError` respectively.

I added tests for both, and fixed the KeyError with a default value.  The exception handler for the Attribute Error is the most efficient option probably anyway.
